### PR TITLE
Add mapreduce for hosts import

### PR DESCRIPTION
**Pull request by wolfgangmeyers**

### CODE REVIEW

**Comment by wolfgangmeyers** [comment]
Yikes! Looks like I won't be able to directly to reply to comments until github fixes their stuff... seeing `github-144a332….js:10 Uncaught TypeError: Cannot read property 'click' of null` whenever I try.

**Comment by wolfgangmeyers** [comment]
@brianmountford A reader must throw NoSuchElementException in order to terminate the stream. If it does not, the reader will continue to emit a stream of null values.

**Comment by wolfgangmeyers** [comment]
@brianmountford Updates made to tests/testdata files.

File: java/google/registry/rde/RdeHostInput.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  private final String importFileName;
+
+  /**
+   * Creates a new {@link RdeHostInput}
+   *
+   * @param mapShards Number of readers that should be created
+   * @param importBucketName Name of GCS bucket for escrow file imports
+   * @param importFileName Name of escrow file in GCS
+   */
+  public RdeHostInput(Optional<Integer> mapShards, String importBucketName,
```
Comment (cornmander) [inline java/google/registry/rde/RdeHostInput.java]: OK

File: java/google/registry/rde/RdeHostReader.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+@NotThreadSafe
+public class RdeHostReader extends InputReader<HostResource> implements Serializable {
+
+  private static final long serialVersionUID = 3037264959150412846L;
+
+  private static final FormattingLogger logger = FormattingLogger.getLoggerForCallerClass();
+  private static final GcsService GCS_SERVICE =
+      GcsServiceFactory.createGcsService(RetryParams.getDefaultInstance());
+
+  final String importBucketName;
```
Comment (cornmander) [inline java/google/registry/rde/RdeHostReader.java]: Sorry, somehow missed that before.

**Comment by brianmountford** [comment]
Closing this pull request. The changes have been integrated into the code base, and will be present in the next open source release.

**Comment by CydeWeys** [comment]
Merged here: https://github.com/google/nomulus/commit/4cfe107475752dd3ebb58a8fc8de73fb6cd5278d

**Pull request closed without merging**